docs/profiling_test.py

```diff
@@ -0,0 +1,120 @@
+import shutil
+
+import pytest
+from omegaconf import DictConfig
+
+from project.conftest import (  # noqa: F401
+    accelerator,
+    algorithm_config,
+    algorithm_network_config,
+    command_line_arguments,
+    datamodule_config,
+    devices,
+    experiment_dictconfig,
+    num_devices_to_use,
+    overrides,
+)
+from project.experiment import setup_experiment
+from project.utils.hydra_utils import resolve_dictconfig
+
+## Direct parametrization would occur with the following code
+# @pytest.fixture
+# def datamodule(command_line_arguments: str) -> LightningDataModule:
+#     # uses hydra to get the datamodule from the command line arguments
+#     # Something like:
+#     # hydra.utils.instantiate(config=command_line_arguments)
+#     # with hydra.utils.setup
+#     ...
+
+
+# @pytest.mark.parametrize("foo", [1, 2, 3])
+# def test_something_about_datamodule(datamodule: LightningDataModule, foo: int):
+#     datamodule.prepare_data()
+#     datamodule.setup("fit")
+#     ...
+
+
+@pytest.mark.skipif(not shutil.which("sbatch"), reason="Needs to be run on a SLURM cluster")
+@pytest.mark.parametrize(
+    "command_line_arguments",
+    [
+        # Instrumenting your code -baseline
+        """
+        experiment=profiling \
+        algorithm=example \
+        trainer.logger.wandb.name="Baseline" \
+        trainer.logger.wandb.tags=["Training","Baseline comparison","CPU/GPU comparison"]
+        """,
+        # Identifying potential bottlenecks - baseline
+        """
+        experiment=profiling\
+        algorithm=no_op\
+        trainer.logger.wandb.name="Baseline without training" \
+        trainer.logger.wandb.tags=["No training","Baseline comparison"]
+
+        """,
+        # Identifying potential bottlenecks - num_workers multirun
+        pytest.param(
+            """
+            -m experiment=profiling \
+            algorithm=no_op \
+            trainer.logger.wandb.tags=["1 CPU Dataloading","Worker throughput"] \
+            datamodule.num_workers=1,4,8,16,32
+            """,
+            marks=pytest.mark.skip(reason="not working"),
```

Comment:
Let's be more precise about why this is failing:
```suggestion
            marks=pytest.mark.xfail(
                reason="Lexer error caused by the -m flag",
                raises=hydra.errors.OverrideParseException,
                strict=True,
            ),
```